code/game/objects/items/spear.dm

```diff
@@ -167,19 +167,17 @@
 	return CLICK_ACTION_SUCCESS
 
 
-/obj/item/spear/explosive/afterattack(atom/movable/AM, mob/user, proximity)
-	. = ..()
-	if(!proximity || !HAS_TRAIT(src, TRAIT_WIELDED) || !istype(AM))
+/obj/item/spear/explosive/afterattack(atom/movable/AM, mob/user, click_parameters)
```

Comment:
both this and the greytide spear mutate this `afterattack()` argument from `atom/target` to `atom/movable/AM` - is that something you want to address in this PR or leave it for another day